### PR TITLE
test: add e2e test to cover PropagationPolicy

### LINKED ISSUE #3884: Add e2e case to cover policy cleanup
Through this issue(#3878), we found that there is a behavior of directly modifying cached resource in the `detector`, we need to add related E2E to cover these scenarios.

related pr: [#3879](https://github.com/karmada-io/karmada/pull/3879)

### PR DESCRIPTION
**Pull request by yike21**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Add e2e case to cover policy cleanup.

**Which issue(s) this PR fixes**:
Fixes #3884 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Ask @whitewindmills to help take a review.
/cc @whitewindmills

**Review by whitewindmills — COMMENTED** [review]
@yike21 thanks~
It doesn't make sense to test the same process with two different resources(`Deployment` and `Configmap`). We need to make sure the following two scenarios are covered:
* After normal propagation, delete policy and create new policy. Check if the new policy is applied.
* After normal propagation,  modify policy to unbind from the resource template and create new policy. Check if the new policy is applied.

**Note: you should add labels to resource templates, otherwise those e2e cases will pass even if the pr https://github.com/karmada-io/karmada/pull/3879 was not merged.**

**Comment by yike21** [comment]
> @yike21 thanks~ It doesn't make sense to test the same process with two different resources(`Deployment` and `Configmap`). We need to make sure the following two scenarios are covered:
> 
> * After normal propagation, delete policy and create new policy. Check if the new policy is applied.
> * After normal propagation,  modify policy to unbind from the resource template and create new policy. Check if the new policy is applied.
> 
> **Note: you should add labels to resource templates, otherwise those e2e cases will pass even if the pr #3879 was not merged.**

I will fix the e2e tests soon, thanks for the clear clarification!

File: test/e2e/propagationpolicy_test.go [hunk 1]
```diff
@@ -1031,4 +1031,88 @@ var _ = ginkgo.Describe("[AdvancedPropagation] propagation testing", func() {
...
+						ClusterNames: []string{member1},
+					},
+				},
+			})
+
+			framework.CreatePropagationPolicy(karmadaClient, policy02)
+			framework.WaitConfigMapDisappearOnCluster(member1, configmap.Namespace, configmap.Name)
+			framework.WaitConfigMapPresentOnClusterFitWith(member2, configmap.Namespace, configmap.Name,
+				func(configmap *corev1.ConfigMap) bool { return true })
+			framework.RemovePropagationPolicy(karmadaClient, policy02.Namespace, policy02.Name)
```
Comment (whitewindmills) [inline test/e2e/propagationpolicy_test.go:1102]: `policy01` should also be removed.

File: test/e2e/propagationpolicy_test.go [hunk 2]
```diff
@@ -1031,4 +1031,88 @@ var _ = ginkgo.Describe("[AdvancedPropagation] propagation testing", func() {
 				func(deployment *appsv1.Deployment) bool { return true })
 		})
 	})
+
+	ginkgo.Context("Unbind PropagationPolicy for configmap and create another", func() {
```
Comment (whitewindmills) [inline test/e2e/propagationpolicy_test.go:1035]: how about `Unbind the old PropagationPolicy and create a new one`?

File: test/e2e/propagationpolicy_test.go [hunk 3]
```diff
@@ -1031,4 +1031,88 @@ var _ = ginkgo.Describe("[AdvancedPropagation] propagation testing", func() {
...
+
+		ginkgo.BeforeEach(func() {
+			framework.CreateConfigMap(kubeClient, configmap)
+			ginkgo.DeferCleanup(func() {
+				framework.RemoveConfigMap(kubeClient, configmap.Namespace, configmap.Name)
+				framework.WaitConfigMapDisappearOnClusters(framework.ClusterNames(), configmap.Namespace, configmap.Name)
+			})
+		})
+
+		ginkgo.It("unbind propagation testing", func() {
```
Comment (whitewindmills) [inline test/e2e/propagationpolicy_test.go:1079]: `modify the old PropagationPolicy to unbind and create a new one`

File: test/e2e/propagationpolicy_test.go [hunk 4]
```diff
@@ -1031,4 +1031,88 @@ var _ = ginkgo.Describe("[AdvancedPropagation] propagation testing", func() {
...
+			})
+
+			framework.CreatePropagationPolicy(karmadaClient, policy02)
+			framework.WaitConfigMapDisappearOnCluster(member1, configmap.Namespace, configmap.Name)
+			framework.WaitConfigMapPresentOnClusterFitWith(member2, configmap.Namespace, configmap.Name,
+				func(configmap *corev1.ConfigMap) bool { return true })
+			framework.RemovePropagationPolicy(karmadaClient, policy02.Namespace, policy02.Name)
+		})
+
+		ginkgo.It("delete propagation testing", func() {
```
Comment (whitewindmills) [inline test/e2e/propagationpolicy_test.go:1105]: `delete the old PropagationPolicy to unbind and create a new one`

File: test/e2e/propagationpolicy_test.go (hunk 1 above)
Comment (yike21) [reply test/e2e/propagationpolicy_test.go:1103]: Thanks! Suggestions have been done.

**Comment by whitewindmills** [comment]
/lgtm
cc @XiShanYongYe-Chang to checkit.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks to all of you.
/approve

**Comment by yike21** [comment]
The CI server seems to be under too much pressure, which causes a timeout [error](https://github.com/karmada-io/karmada/actions/runs/5794608606/job/15704816413?pr=3904#step:5:1818). And from [here](https://github.com/yike21/karmada/actions/runs/5794608000) we can see that the corresponding e2e test have passed. Maybe it needs another try. 
Please help restart CI, thanks! @RainbowMango

**Comment by RainbowMango** [comment]
done

**Review by whitewindmills — COMMENTED** [review]
/lgtm

**Pull request merged**